src/axom/primal/operators/detail/intersect_curved_poly_impl.hpp

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ * \file intersect.hpp
+ *
+ * \brief Consists of functions to test intersection among geometric primitives.
+ */
+
+#ifndef INTERSECTION_CURVED_POLYGON_HPP_
+#define INTERSECTION_CURVED_POLYGON_HPP_
+
+#include "axom/primal/geometry/BoundingBox.hpp"
+#include "axom/primal/geometry/OrientedBoundingBox.hpp"
+#include "axom/primal/geometry/Point.hpp"
+#include "axom/primal/geometry/Ray.hpp"
+#include "axom/primal/geometry/Segment.hpp"
+#include "axom/primal/geometry/Sphere.hpp"
+#include "axom/primal/geometry/Triangle.hpp"
+#include "axom/primal/geometry/BezierCurve.hpp"
+#include "axom/primal/geometry/CurvedPolygon.hpp"
+
+#include "axom/core/utilities/Utilities.hpp"
+
+#include "axom/primal/operators/squared_distance.hpp"
+#include "axom/primal/operators/detail/intersect_bezier_impl.hpp"
+
+namespace axom
+{
+namespace primal
+{
+namespace detail
+{
+template < typename T >
+class IntersectionInfo;
+
+template <typename T>
+bool orient(  const BezierCurve<T,2>& c1, 
+              const BezierCurve<T,2>& c2, 
+              T s, 
+              T t);
+
+template <typename T>
+int isContained(  const CurvedPolygon<T,2>& p1, 
+                  const CurvedPolygon<T,2>& p2, 
+                  double sq_tol = 1e-10);
+
+/*!
+ * \brief Test whether the regions within CurvedPolygons p1 and p2 intersect. 
+ * \return status true iff p1 intersects with p2, otherwise false.
+ *
+ * \param [in] p1, p2 CurvedPolygon objects to intersect
+ * \param [in] sq_tol tolerance parameter for the base case of intersect_bezier_curve
+ * \param [out] pnew vector of type CurvedPolygon holding CurvedPolygon objects representing boundaries of intersection regions. 
+ */
+template < typename T, int NDIMS>
+bool intersect_polygon(  const CurvedPolygon< T, NDIMS>& p1,
+                         const CurvedPolygon< T, NDIMS>& p2,
+                         std::vector<CurvedPolygon< T, NDIMS>>& pnew,
+                         double sq_tol
+                       )
+{
+  // Object to store intersections
+  std::vector<std::vector<IntersectionInfo<T>>> E1IntData(p1.numEdges());
```

Comment:
@davidgunderman -- I think the macos problem is getting triggered here. 
For some reason, xcode is not using ``IntersectionInfo<T>::operator<()`` to sort.

```cpp
   std::vector<std::vector<IntersectionInfo<T>>> E1IntData(p1.numEdges()); 
   std::vector<std::vector<IntersectionInfo<T>>> E2IntData(p2.numEdges()); 
   ...
   std::sort( E1IntData[i].begin(), E1IntData[i].end() );
   std::sort( E2IntData[i].begin(), E2IntData[i].end() ); 
```

Reply:
I'm really not sure why this would be happening. I'll spend some time today looking around.

Reply:
@kennyweiss I think this is fixed now. It looks like custom operator< definition needed to be const and/or needed to take a reference as a parameter

Reply:
@kennyweiss I think this is fixed now.